minio/credentials/providers.py

```diff
@@ -187,7 +187,11 @@ def retrieve(self):
 class IAMProvider(Provider):
     """IAM EC2 credential provider."""
 
-    def __init__(self, endpoint=None, http_client=None):
+    def __init__(self,
+                 endpoint=None,
+                 http_client=None,
+                 expiry_delta=None):
```

Comment:
Maybe it is not that useful to add expiry_delta parameter here. Let's negate 10 seconds by default unless we hear something from the community (I don't think we will)

Reply:
Yes it is 10sec inside the code @vadmeste as an if condition rather than defaulted at the constructor.

Reply:
> Yes it is 10sec inside the code @vadmeste as an if condition rather than defaulted at the constructor.

I am just saying, let's not expose this variable to users. I don't see how this is useful. We can make it 10 seconds by default. (LGTM otherwise)

Reply:
It is necessary for situations where renewal can be aggressive or faster.  It also complies with minio-go behavior. So a configurable value is not a problem.